code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,201 @@
+/* 	< OH MY GOD. Can't you just make "/image/proc/foo()" instead of making these? >
+ * 		/appearance is a hardcoded byond type, and it is very internal type.
+ *		Its type is actually /image, but it isn't truly /image. We defined it as "/appearance"
+ * 		new procs to /image will only work to actual /image references, but...
+ * 		/appearance references are not capable of executing procs, because these are not real /image
+ * 		This is why these global procs exist. Welcome to the curse.
+ */
+/// Makes a var list of /appearance type actually uses. This will be only called once.
+/proc/build_virtual_appearance_vars()
+	. = list("vis_flags") // manual listing.
+	var/list/unused_var_names = list(
+		"appearance", // it only does self-reference
+		"x","y","z", // these are always 0
+		"weak_reference", // it's not a good idea to make a weak_ref on this, and this won't have it
+		"vars", // inherited from /image, but /appearance hasn't this
+
+		// Even if these vars are essential for image, these only exists in an actual type
+		"filter_data", 
+		"realized_overlays",
+		"realized_underlays",
+				
+		// we have no reason to show these, right?
+		"_active_timers",
+		"_datum_components",
+		"_listen_lookup",
+		"_signal_procs",
+		"__auxtools_weakref_id",
+		"_status_traits",
+		"cooldowns",
+		"datum_flags",
+		"visibility",
+		"verbs",
+		"gc_destroyed",
+		"harddel_deets_dumped",
+		"open_uis",
+		"tgui_shared_states"
+		)
```

Comment:
Is there a way you can use `NAMEOF()` in this list? it's a really nice byond voodoo helper that will really help us out in the event something like `visibility` doesn't exist any more on datums- it'll trigger a compile failure and whoever's removing `visibility` will be forced to remove it from this list and keep it nice and maintained.

(we already use `NAMEOF()` a lot in VV anyways)

Reply:
It'll be another janky code with it, I think...
```DM
if(NAMEOF(D, datum_flags) || TRUE) unused_var_names += "datum_flags"
```
or
```DM
#define NAMEOF_APPEARANCE(datum, X, _list) if(NAMEOF(D, X) || TRUE) _list += #X
NAMEOF_APPEARANCE(image_ref, datum_flags, unused_var_names)
```
Maybe worth doing it because it will allow us notice if any variable name is changed, but I can't think a good looking format.

Reply:
I think I will keep this as is too.
Adding such macro can make people read it hardly, and this is simple enough to understand, even with a very minor risk that we'll miss when /datum has no longer `visibility` - a minor risk that nobody will remove "visibility" manually from here, but would ever anyone care? or, if maintained, this raw text will be easier to read.